src/docker/manifest.nim

```diff
@@ -310,34 +280,33 @@ proc fetchManifest*(name: DockerImage, otherNames: seq[DockerImage] = @[]): Dock
       meta = manifestHead(name)
       data = manifestGet(name.withDigest(meta.digest), meta.mediaType)
     result = newManifest(name, data, otherNames = otherNames)
-    result.fetch()
   except RegistryResponseError:
     trace("docker: " & getCurrentExceptionMsg())
     raise
   except:
     error("docker: " & getCurrentExceptionMsg())
     let data = requestManifestJson(name)
     result = newManifest(name, data, otherNames = otherNames)
-    result.fetch()
+  result.fetch(fetchConfig = fetchConfig)
   manifestCache[name.asRepoDigest()] = result
 
-proc fetchOnlyImageManifest*(name: DockerImage): DockerManifest =
-  var manifest = fetchManifest(name)
+proc fetchOnlyImageManifest*(name: DockerImage, fetchConfig = true): DockerManifest =
+  var manifest = fetchManifest(name, fetchConfig = fetchConfig)
   if manifest.kind == DockerManifestType.list:
     let manifests = manifest.findAllPlatformsManifests()
     if len(manifests) == 1:
       manifest = manifests[0]
-      manifest.fetch()
     else:
       raise newException(KeyError, "There are multiple platform images for: " & $name)
```

Comment:
Would adding a platform arg here, defaulting to the host platform, alleviate the multiple platform image exception?

Reply:
no

this is used in https://github.com/crashappsec/chalk/blob/84d24ad22bbbfa3ecd336e7bdf412938ed98398d/src/docker/platform.nim#L107 which is inside https://github.com/crashappsec/chalk/blob/84d24ad22bbbfa3ecd336e7bdf412938ed98398d/src/docker/platform.nim#L75-L76 which is required for https://github.com/crashappsec/chalk/blob/84d24ad22bbbfa3ecd336e7bdf412938ed98398d/src/docker/platform.nim#L265-L269

there are a few fallbacks in order to correctly determine correct platform of the build. it does not always match host platform.

for example:

```dockerfile
FROM foo:arm64
CMD foo
```

this will build arm64 image even if the host platform is amd64. what `foo:arm64` is has 2 possibilities though:

* its a manifest list which has a single platform image inside of it. for example this will happen when enabling provenance but not doing multi-platform build. it will yield manifest list with a single build in it where the second reference will be attestation itself
* its not a manifest list but an image manifest directly which will have platform specified

code for that: https://github.com/crashappsec/chalk/blob/84d24ad22bbbfa3ecd336e7bdf412938ed98398d/src/docker/platform.nim#L105-L114

so in this case we dont care the platform itself, just care that there is a single platform in the manifest list

Reply:
clear. thank you.